src/tbb/market.cpp

```diff
@@ -364,8 +364,16 @@ arena* market::arena_in_need ( arena_list_type* arenas, arena* hint ) {
             } while ( arenas[curr_priority_level].empty() );
             it = arenas[curr_priority_level].begin();
         }
-        if( a.num_workers_active() < a.my_num_workers_allotted.load(std::memory_order_relaxed) ) {
-            a.my_references += arena::ref_worker;
+
+        auto try_join = [&] {
```

Comment:
Any particular reason to use a lambda here? Can't this just be written in the subsequent if, with an additional if in the body?

Reply:
Applied